beater/route_config.go

```diff
@@ -79,37 +79,60 @@ var V1Routes = map[string]v1Route{
 
 var V2Routes = map[string]v2Route{
 	V2BackendURL: v2BackendRoute,
-	V2RumURL:     {rumRouteType},
+	V2RumURL:     v2RumRoute,
 }
 
-var v2BackendRoute = v2Route{
-	routeType{
-		v2backendHandler,
-		systemMetadataDecoder,
-		func(*Config) transform.Config { return transform.Config{} },
-	},
-}
+var (
+	v2BackendRoute = v2Route{
+		routeType{
+			v2backendHandler,
+			systemMetadataDecoder,
+			func(*Config) transform.Config { return transform.Config{} },
+			nilCache,
+		},
+	}
+	v2RumRoute = v2Route{
+		routeType{
+			v2rumHandler,
+			userMetaDataDecoder,
+			rumTransformConfig,
+			func(c *Config) *rlCache {
```

Comment:
sorry, I've changed my mind regarding this. It was generally the idea that the `routeType`s would contain all the stuff that is common for v1 and v2 while `v2Route` allows special config for v2. I think i forgot that when we last discussed this. I think the rlCache should be part of `v2Route` instead because rlCache is only used in v2.